tests/ctst/common/common.ts

```diff
@@ -357,32 +358,23 @@ Then('kafka consumed messages should not take too much place on disk', { timeout
 
                 for (let i = 0; i < topics.length; i++) {
                     this.logger.debug('Checking topic', { topic: topics[i] });
+                    let topicCleaned = true;
                     for (let j = 0; j < newOffsets[i].partitions.length; j++) {
-                        const newMessagesAfterClean =
-                            newOffsets[i].partitions[j].low === previousOffsets[i].partitions[j].high &&
-                            newOffsets[i].partitions[j].high > previousOffsets[i].partitions[j].high;
-
-                        if (newMessagesAfterClean) {
-                            // If new messages appeared after we gathered the offsets, we need to recheck after
-                            this.logger.warn('New messages after clean', { topic: topics[i] });
-                            continue;
-                        }
+                        const partition = newOffsets[i].partitions[j];
 
-                        const lowOffsetIncreased = newOffsets[i].partitions[j].low >
-                            previousOffsets[i].partitions[j].low;
-                        const allMessagesCleaned = newOffsets[i].partitions[j].high ===
-                            newOffsets[i].partitions[j].low;
-
-                        // If the low offset increased it means the topic has been cleaned
-                        // If low offset is the same as high offset,
-                        // it means the topic is completly cleaned even though lowOffset didnt increased
-                        assert.ok(lowOffsetIncreased || allMessagesCleaned,
-                            `Topic ${topics[i]} partition ${j} offset has not increased,
-                            previousOffsets: ${previousOffsets[i].partitions[j].low} /\
-                            ${previousOffsets[i].partitions[j].high},
-                            newOffsets: ${newOffsets[i].partitions[j].low} / ${newOffsets[i].partitions[j].high}`);
-
-                        // Topic is cleaned, we don't need to check it anymore
+                        // Ensure we're accessing the correct partition details
+                        const lowOffsetIncreased = parseInt(partition.low) >
```

Comment:
why do we need to `parseInt` here and not below?
Why don't we create a variable as well for previousOffset ?

Reply:
if the PR is in draft state, you should keep it in draft ; and you can request integration branches by explicitely sending the `/create_integration_branches` command to bert-e